db/migrate/20190703130101_move_recruitment_cycle_to_provider.rb

```diff
@@ -0,0 +1,73 @@
+require_relative '20190619142449_add_recruitment_cycle_to_course'
+require_relative '20190619144634_add_2019_recruitment_cycle_to_courses'
+require_relative '20190620060536_set_recruitment_cycle_to_not_null'
+require_relative '20190620120944_add_recruitment_cycle_to_site'
+require_relative '20190620121519_add_2019_recruitment_cycle_to_sites'
+require_relative '20190621125905_set_site_recruitment_cycle_to_not_null'
+
+class MoveRecruitmentCycleToProvider < ActiveRecord::Migration[5.2]
+  def up
+    add_reference :provider, :recruitment_cycle, index: false, type: :int, foreign_key: true
+
+    current_recruitment_cycle = RecruitmentCycle.where(year: '2019').first
+
+    say_with_time 'updating provider recruitment cycle' do
+      Provider.connection.update <<~EOSQL
+        UPDATE provider SET recruitment_cycle_id=#{current_recruitment_cycle.id}
+      EOSQL
+    end
+
+    say_with_time 'SetSiteRecruitmentCycleToNotNull' do
+      revert SetSiteRecruitmentCycleToNotNull
```

Comment:
I'm assuming this and all the other migrations in here won't be re-applied the next time migrations are run?

Reply:
Ya, it just runs the down migrations for these, but doesn't remove them from the `schema_migrations`. That means it knows how to rollback this migration, by redoing these ones. We've tested going back and forth a few times.